crucible-llvm/src/Lang/Crucible/LLVM/MemModel/Type.hs

```diff
@@ -139,8 +137,19 @@ typeEnd a tp = seq a $
     X86_FP80 -> a + 10
     Array 0 _   -> a
     Array n etp -> typeEnd (a + natBytesMul (n-1) (storageTypeSize etp)) etp
-    Struct flds -> typeEnd (a + fieldOffset f) (f^.fieldVal)
-      where f = V.last flds
+    Struct flds ->
+      if V.length flds == 0
+      then 0
+      else
+        let f = V.last flds
+        in typeEnd (a + fieldOffset f) (f^.fieldVal)
+
+-- | Returns end of field including padding bytes.
+structSize :: V.Vector (Field StorageType) -> Bytes
+structSize flds =
+  if V.length flds == 0
+  then 0
+  else fieldEnd (V.last flds)
```

Comment:
I'm a bit leery about using partial functions like `last`. You could rewrite this function to be total using [`unsnoc`](https://hackage.haskell.org/package/vector-0.12.3.1/docs/Data-Vector.html#v:unsnoc), which should also be constant time.

Reply:
Nice, that's much better. Thanks!